sway/input/cursor.c

```diff
@@ -162,7 +162,29 @@ void cursor_send_pointer_motion(struct sway_cursor *cursor, uint32_t time_msec,
 				seat_set_focus_warp(cursor->seat, c, false);
 			}
 		} else {
-			seat_set_focus_warp(cursor->seat, c, false);
+			// Don't switch focus on title mouseover for
+			// stacked and tabbed layouts
+			// If pointed container is in nested containers which are
+			// inside tabbed/stacked layout we should skip them
+			bool do_mouse_focus = true;
+			struct sway_container *p = c->parent;
+			while (p) {
+				if ((p->layout == L_TABBED || p->layout == L_STACKED)
+					&& !view_is_visible(c->sway_view)) {
+					do_mouse_focus = false;
+					break;
+				}
+				p = p->parent;
+			}
+			if (!do_mouse_focus) {
+				struct sway_container *next_focus = seat_get_focus_inactive(
+						cursor->seat, p);
+				if (next_focus && view_is_visible(next_focus->sway_view)) {
```

Comment:
Is this `view_is_visible` check needed? If so, assert that `next_focus` is a view. If not, remove it.